diff --git a/frameworks/js-bindings/cocos2d-x/cocos/2d/CCActionManager.cpp b/frameworks/js-bindings/cocos2d-x/cocos/2d/CCActionManager.cpp
index 0fb432a..e2d15c1 100644
--- a/frameworks/js-bindings/cocos2d-x/cocos/2d/CCActionManager.cpp
+++ b/frameworks/js-bindings/cocos2d-x/cocos/2d/CCActionManager.cpp
@@ -312,7 +312,7 @@ Action* ActionManager::getActionByTag(int tag, const Node *target) const
                 }
             }
         }
-        CCLOG("cocos2d : getActionByTag(tag = %d): Action not found", tag);
+        //CCLOG("cocos2d : getActionByTag(tag = %d): Action not found", tag);
     }
     else
     {
diff --git a/frameworks/js-bindings/cocos2d-x/cocos/platform/mac/CCDevice.mm b/frameworks/js-bindings/cocos2d-x/cocos/platform/mac/CCDevice.mm
index 0d13a3a..4cb0460 100644
--- a/frameworks/js-bindings/cocos2d-x/cocos/platform/mac/CCDevice.mm
+++ b/frameworks/js-bindings/cocos2d-x/cocos/platform/mac/CCDevice.mm
@@ -118,8 +118,7 @@ static bool _initWithString(const char * text, Device::TextAlign align, const ch
 				NSMutableString *lineBreak = [[[NSMutableString alloc] init] autorelease];
 				NSUInteger length = [string length];
 				NSRange range = NSMakeRange(0, 1);
-                CGSize textSize;
-				NSUInteger lastBreakLocation = 0;
+                NSUInteger lastBreakLocation = 0;
                 NSUInteger insertCount = 0;
 				for (NSUInteger i = 0; i < length; i++) {
 					range.location = i;
@@ -128,7 +127,7 @@ static bool _initWithString(const char * text, Device::TextAlign align, const ch
 					if ([@"!?.,-= " rangeOfString:character].location != NSNotFound) {
                         lastBreakLocation = i + insertCount;
                     }
-                    textSize = [lineBreak sizeWithAttributes:tokenAttributesDict];
+                    auto textSize = [lineBreak sizeWithAttributes:tokenAttributesDict];
                     if(textSize.height > info->height)
                         break;
 					if (textSize.width > info->width) {
diff --git a/frameworks/js-bindings/cocos2d-x/extensions/GUI/CCEditBox/CCEditBoxImplMac.h b/frameworks/js-bindings/cocos2d-x/extensions/GUI/CCEditBox/CCEditBoxImplMac.h
index a5b93c4..d7abaff 100644
--- a/frameworks/js-bindings/cocos2d-x/extensions/GUI/CCEditBox/CCEditBoxImplMac.h
+++ b/frameworks/js-bindings/cocos2d-x/extensions/GUI/CCEditBox/CCEditBoxImplMac.h
@@ -39,6 +39,8 @@
 
 @interface CCEditBoxImplMac : NSObject <NSTextFieldDelegate>
 {
+    NSTextField* textField_;
+	NSSecureTextField *secureTextField_;
     void* editBox_;
     BOOL editState_;
     NSMutableDictionary* placeholderAttributes_;
diff --git a/frameworks/js-bindings/cocos2d-x/extensions/GUI/CCEditBox/CCEditBoxImplMac.mm b/frameworks/js-bindings/cocos2d-x/extensions/GUI/CCEditBox/CCEditBoxImplMac.mm
index fbeed2e..28e7a56 100644
--- a/frameworks/js-bindings/cocos2d-x/extensions/GUI/CCEditBox/CCEditBoxImplMac.mm
+++ b/frameworks/js-bindings/cocos2d-x/extensions/GUI/CCEditBox/CCEditBoxImplMac.mm
@@ -324,7 +324,7 @@ void EditBoxImplMac::setPlaceholderFont(const char* pFontName, int fontSize)
         return;
     }
     
-    _sysEdit.placeholderAttributes[NSFontAttributeName] = font;
+    [_sysEdit.placeholderAttributes setObject:font forKey:NSFontAttributeName];
     
     /* reload placeholder */
     const char *placeholder = [_sysEdit.textField.cell placeholderAttributedString].string.UTF8String;
@@ -343,7 +343,7 @@ void EditBoxImplMac::setFontColor(const Color3B& color)
 void EditBoxImplMac::setPlaceholderFontColor(const Color3B& color)
 {
     NSColor *nsColor = [NSColor colorWithCalibratedRed:color.r/255.f green:color.g / 255.f blue:color.b / 255.f alpha:1.0f];
-    _sysEdit.placeholderAttributes[NSForegroundColorAttributeName] = nsColor;
+    [_sysEdit.placeholderAttributes setObject:nsColor forKey:NSForegroundColorAttributeName];
     
     /* reload placeholder */
     const char *placeholder = [_sysEdit.textField.cell placeholderAttributedString].string.UTF8String;
diff --git a/frameworks/js-bindings/cocos2d-x/plugin/jsbindings/script/jsb_pluginx.js b/frameworks/js-bindings/cocos2d-x/plugin/jsbindings/script/jsb_pluginx.js
index 250b807..cb1901a 100644
--- a/frameworks/js-bindings/cocos2d-x/plugin/jsbindings/script/jsb_pluginx.js
+++ b/frameworks/js-bindings/cocos2d-x/plugin/jsbindings/script/jsb_pluginx.js
@@ -1,4 +1,4 @@
-plugin = plugin || {};
+var plugin = plugin || {PluginParam: {}, ProtocolAds: {}, ProtocolIAP: {}, ProtocolShare: {}, ProtocolSocial: {}, ProtocolUser: {}};
 
 plugin.PluginParam.ParamType = {};
 plugin.PluginParam.ParamType.TypeInt = 1;
