diff --git a/cocos/base/ccConfig.h b/cocos/base/ccConfig.h
index 0d32b51..d6cd15f 100644
--- a/cocos/base/ccConfig.h
+++ b/cocos/base/ccConfig.h
@@ -255,14 +255,14 @@ THE SOFTWARE.
 /** Use 3d physics integration API. */
 #ifndef CC_USE_3D_PHYSICS
 #if (CC_TARGET_PLATFORM == CC_PLATFORM_IOS || CC_TARGET_PLATFORM == CC_PLATFORM_MAC || CC_TARGET_PLATFORM == CC_PLATFORM_WIN32 || CC_TARGET_PLATFORM == CC_PLATFORM_ANDROID || CC_TARGET_PLATFORM == CC_PLATFORM_LINUX || CC_TARGET_PLATFORM == CC_PLATFORM_WINRT)
-#define CC_USE_3D_PHYSICS 1
-#define CC_USE_3D_PHYSICS 1
+#define CC_USE_3D_PHYSICS 0
+#define CC_USE_3D_PHYSICS 0 
 #endif
 #endif
 
 /** Use 3D navigation API */
 #ifndef CC_USE_NAVMESH
-#define CC_USE_NAVMESH 1
+#define CC_USE_NAVMESH 0
 #endif
 
 /** Use culling or not. */
